### PR TITLE
Add validations webhook for Tinkerbell Datacenter config

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Issue #, if available:*

*Description of changes:*
Add validations webhook for Tinkerbell Datacenter config
*Testing (if applicable):*
- unit tests
- manual test with tilt

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by taneyland — COMMENTED** [review]
:raised_hands:

File: pkg/api/v1alpha1/tinkerbelldatacenterconfig_webhook.go [hunk 1]
```diff
@@ -42,15 +46,49 @@ var _ webhook.Validator = &TinkerbellDatacenterConfig{}
...
+		return apierrors.NewInvalid(
+			GroupVersion.WithKind(TinkerbellDatacenterKind).GroupKind(),
+			r.Name,
+			field.ErrorList{
+				field.Invalid(field.NewPath("spec"), r.Spec, err.Error()),
+			},
+		)
+	}
+
+	if r.IsReconcilePaused() {
```
Comment (cxbrowne1207) [inline pkg/api/v1alpha1/tinkerbelldatacenterconfig_webhook.go:59]: Do we also need to validate that the machine config has a template defined like with vSphere?

Comment (mitalipaygude) [reply pkg/api/v1alpha1/tinkerbelldatacenterconfig_webhook.go:59]: This webhook is for the datacenter config for Tinkerbell, not the machine config. We can have that check for the machine config webhook.

File: pkg/providers/tinkerbell/validate.go [hunk 2]
```diff
@@ -3,47 +3,16 @@ package tinkerbell
...
 	tinkv1alpha1 "github.com/tinkerbell/tink/pkg/apis/core/v1alpha1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
-	"github.com/aws/eks-anywhere/pkg/logger"
 	"github.com/aws/eks-anywhere/pkg/networkutils"
 	"github.com/aws/eks-anywhere/pkg/providers/tinkerbell/hardware"
 )
 
-func validateDatacenterConfig(config *v1alpha1.TinkerbellDatacenterConfig) error {
```
Comment (g-gaston) [inline pkg/providers/tinkerbell/validate.go:18]: I thought the providers team wanted to keep all validation in the `tinkerbell` package?
I'm totally ok with this, I have no concerns, just making sure everyone agrees it.

File: pkg/api/v1alpha1/tinkerbelldatacenterconfig.go [hunk 3]
```diff
@@ -40,3 +47,41 @@ func GetTinkerbellDatacenterConfig(fileName string) (*TinkerbellDatacenterConfig
 	}
 	return &clusterConfig, nil
 }
+
+func validateDatacenterConfig(config *TinkerbellDatacenterConfig) error {
+	if config.Spec.OSImageURL != "" {
+		if _, err := url.ParseRequestURI(config.Spec.OSImageURL); err != nil {
+			return fmt.Errorf("parsing osImageOverride: %v", err)
+		}
+		logger.Info("OS image path override set", "path", config.Spec.OSImageURL)
```
Comment (g-gaston) [inline pkg/api/v1alpha1/tinkerbelldatacenterconfig.go:56]: Why is there a log in the validations?

File: pkg/api/v1alpha1/tinkerbelldatacenterconfig.go [hunk 4]
```diff
@@ -40,3 +47,41 @@ func GetTinkerbellDatacenterConfig(fileName string) (*TinkerbellDatacenterConfig
...
+	}
+
+	if config.Spec.HookImagesURLPath != "" {
+		if _, err := url.ParseRequestURI(config.Spec.HookImagesURLPath); err != nil {
+			return fmt.Errorf("parsing hookOverride: %v", err)
+		}
+		logger.Info("hook path override set", "path", config.Spec.HookImagesURLPath)
```
Comment (g-gaston) [inline pkg/api/v1alpha1/tinkerbelldatacenterconfig.go:63]: same here about logs

File: pkg/providers/tinkerbell/validate.go (hunk 2 above)
Comment (mitalipaygude) [reply pkg/providers/tinkerbell/validate.go:18]: Yes we discussed and agreed on this approach.

File: pkg/api/v1alpha1/tinkerbelldatacenterconfig_webhook.go (hunk 1 above)
Comment (cxbrowne1207) [reply pkg/api/v1alpha1/tinkerbelldatacenterconfig_webhook.go:59]: Okay, thanks. I misread the file name!

**Pull request merged**